hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/client/impl/DfsClientConf.java

```diff
@@ -290,6 +291,16 @@ public DfsClientConf(Configuration conf) {
     leaseHardLimitPeriod =
         conf.getLong(HdfsClientConfigKeys.DFS_LEASE_HARDLIMIT_KEY,
             HdfsClientConfigKeys.DFS_LEASE_HARDLIMIT_DEFAULT) * 1000;
+
+    clientShortCircuitNum = conf.getInt(
+        HdfsClientConfigKeys.DFS_SHORT_CIRCUIT_NUM,
+        HdfsClientConfigKeys.DFS_SHORT_CIRCUIT_NUM_DEFAULT);
+    Preconditions.checkArgument(clientShortCircuitNum >= 1,
+        HdfsClientConfigKeys.DFS_SHORT_CIRCUIT_NUM +
+        "can't be less then 1.");
+    Preconditions.checkArgument(clientShortCircuitNum <= 5,
+        HdfsClientConfigKeys.DFS_SHORT_CIRCUIT_NUM +
```

Comment:
any reason why this is limited to 5?

Reply:
I'm not sure but afraid if someone has slowly CPU and set some big value (10 for example) it may overload CPU and negative affect to other processes. The second reason, after 5 increasing performance quite small.